cpp/src/search/contains_table.cu

```diff
@@ -36,6 +36,60 @@ namespace {
 using cudf::experimental::row::lhs_index_type;
 using cudf::experimental::row::rhs_index_type;
 
+/**
+ * @brief An adapter functor to support strong index types for row hasher.
+ */
+template <typename Hasher>
+struct strong_index_hasher_adapter {
+  strong_index_hasher_adapter(Hasher const& hasher) : _hasher{hasher} {}
+
+  template <typename T>
+  __device__ inline auto operator()(T const idx) const noexcept
+  {
+    return _hasher(static_cast<size_type>(idx));
```

Comment:
Will there be narrow conversion issues?

Reply:
This functor is only used locally in this file, and the input `idx` is only strong index types (either `lhs_index_type` or `rhs_index_type`).